jac-uk/admin
# Feature/2560 update phone number fields

## What's included?
The phone number field is now readonly in both the Application and Candidate form in Admin.
The mobile field has been added which validates for a UK mobile number only, ie landline or international formatted numbers are invalid.
Sets the mobileVerifiedAt field to null so when the mobile number is changed the candidate will be forced to verify the number (though this is part of a separate ticket).

Closes #2560 

## Who should test?
✅ Product owner
✅ Developers
✅ UTG

## How to test?

For the tests below you will be modifying data in an application and the candidate's profile page.

**Test 1**
Go to the following URL: https://jac-admin-develop--pr2594-feature-2560-update-rgjg76oh.web.app/
- Find an application in an exercise and make a note of the link to the application
- Click the Edit button
- See that the phone field cannot be edited
- Click to change the mobile field
- Try to enter some invalid numbers and see an error message display (+447971596199, 01272203729, etc)
- Enter a valid UK mobile number, eg 07971596199 and press 'Save' - MAKE A NOTE OF THIS NUMBER
- The number will update with no error

**Test 2**
- Go to the candidate page for the user in the application you used above
- Ensure the mobile number here matches the one you entered in the previous section!
- Click the Edit button
- See that the phone field cannot be edited
- Click to change the mobile field
- Try to enter some invalid numbers and see an error message display (+447971596199, 01272203729, etc)
- Enter a valid UK mobile number, eg 07971596199 and press 'Save' - MAKE A NOTE OF THIS NUMBER
- The number will update with no error

**Test 3**
- Go back to the application you used above
- Check that the mobile number is different from the one you entered in the candidate profile

## Resources/Screenshots/Videos


https://github.com/user-attachments/assets/8b1e474b-50a0-402d-9b38-17cbb8f3caa5



## Risk - how likely is this to impact other areas?
🟢 No risk - this is a self-contained piece of work

## Related permissions
Have permissions been considered for this functionality?
- No permission changes required

---
PREVIEW:DEVELOP
_can be OFF, DEVELOP or STAGING_
